src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -94,4 +109,89 @@ private static String upperFirstLetter(String s) {
   public String getFieldGetFunctionName(TypeRef type, Name identifier) {
     return methodName(identifier);
   }
+
+  public String getGrpcClientTypeName() {
+    // getNickname() thinks the client is a message type and appends a "*".
+    // substring gets rid of the star.
+    return converter.getTypeName(service).getNickname().substring(1) + "Client";
+  }
+
+  public String getGrpcClientConstructorName() {
+    return getGrpcClientTypeName().replace(".", ".New");
+  }
+
+  public String getCallOptionsName() {
+    return clientNamePrefix() + "CallOptions";
+  }
+
+  public String getDefaultClientOptionFunc() {
+    return "default" + getClientName() + "Options";
+  }
+
+  public String getDefaultCallOptionFunc() {
+    return "default" + getCallOptionsName();
+  }
+
+  public String getClientName() {
+    return clientNamePrefix() + "Client";
+  }
+
+  public String getClientConstructorName() {
+    return "New" + getClientName();
```

Comment:
Mostly done.

I still have `getGrpcClientTypeName` and `getGrpcClientConstructorName` doing raw string manipulation. We get the string by `converter.getTypeName`, and the name has a dot. Is this something we want to do in `Name`? I think generalizing just for this exception isn't worth it.